core/network/helpers/protobuf_message_read_writer.hpp

```diff
@@ -16,6 +16,7 @@
 #include "network/adapters/protobuf.hpp"
 #include "network/adapters/uvar.hpp"
 #include "network/helpers/message_read_writer.hpp"
+#include "network/helpers/compressor/compressor.h"
```

Comment:
Instead of mixing protobuf and compression,
you can compose them
```
struct ZstdMessageReadWriter : MessageReadWriter {
  ZstdMessageReadWriter(MessageReadWriter);
};
struct ProtobufMessageReadWriter {
  ProtobufMessageReadWriter(MessageReadWriter);
};
auto state_sync_read_writer =
  ProtobufMessageReadWriter(
    ZstdMessageReadWriter(
      MessageReadWriterUvarint(
        stream)));
```